### PR TITLE
Merge a part of reference templates to related resource templates

### PR DESCRIPTION
**Pull request by ACMoretxj**
* Replace a set of reference including statements in main templates with resource templates.
* Add several new resource templates, which contained in mapping spreadsheets while not revealed before.
* Add comments in the front of template resource templates introducing reference varaibles used in this template.
* Remove condensed reference templates (approximately 31 files).

### CODE REVIEW

**Review by ranvijaykumar — CHANGES_REQUESTED** [review]

File: data/Templates/Hl7v2/Resource/_Bundle.liquid [hunk 1]
```diff
@@ -8,8 +8,8 @@
         {% endif -%}
         {% comment -%} Placeholder provided for customization by users. If left empty, it will be removed in post-processing. {% endcomment -%}
         "identifier":
-        [
-        ],
+        {
```
[the remark this answers is not in the document]
Comment (ACMoretxj) [reply data/Templates/Hl7v2/Resource/_Bundle.liquid:11]: It seems that the `Bundle.identifier` isn't an array according to the [FHIR spec](http://build.fhir.org/bundle.html).

File: data/Templates/Hl7v2/Resource/_AllergyIntolerance.liquid [hunk 2]
```diff
@@ -1,3 +1,12 @@
+{% comment -%}
+The following are references ID accepted by this template.
```
[the remark this answers is not in the document]
Comment (ACMoretxj) [reply data/Templates/Hl7v2/Resource/_AllergyIntolerance.liquid]: I've updated this sentence, thanks.

**Comment by BoyaWu10** [comment]
Let's make sure we have the same changes made in the generation tool.

**Review by BoyaWu10 — APPROVED** [review]

**Review by ACMoretxj — COMMENTED** [review]
The description sentences are adjusted.

**Review by ranvijaykumar — APPROVED** [review]

**Pull request merged**